File: frontend/src/domain/types/controlObjective.ts
import type { Float, Integer } from 'type-fest'

export type ControlObjective = {
  controlPriorityLevel: Float<number>
  facade: string
  id: Integer<number>
  segment: string
  targetNumberOfControlsAtPort: Integer<number>
  targetNumberOfControlsAtSea: Integer<number>
  year: Integer<number>
}

export type UpdateControlObjective = {
  controlPriorityLevel: Float<number> | null
  targetNumberOfControlsAtPort: Integer<number> | null
  targetNumberOfControlsAtSea: Integer<number> | null
}

File: backend/src/main/kotlin/fr/gouv/cnsp/monitorfish/domain/entities/control_objective/ControlObjective.kt
package fr.gouv.cnsp.monitorfish.domain.entities.control_objective

data class ControlObjective(
    val id: Int? = null,
    val facade: String,
    val segment: String?,
    val year: Int?,
    val targetNumberOfControlsAtSea: Int,
    val targetNumberOfControlsAtPort: Int,
    val controlPriorityLevel: Double,
)
